gossip/gossip/algo/pull.go

```diff
@@ -319,10 +321,28 @@ func (engine *PullEngine) OnRes(items []string, nonce uint64) {
 }
 
 func (engine *PullEngine) newNONCE() uint64 {
-	n := uint64(0)
+	engine.lock.Lock()
```

Comment:
we don't need to lock at all. We only create nonces from [here](https://github.com/hyperledger/fabric/blob/0501b0ddcfca9992114a3e137ed5b05156c7e13f/gossip/gossip/algo/pull.go#L179) and we already lock there. 

Just loop until `engine.outgoingNONCES.Exists(n)` is false, no need for attempts or anything.